Client/Application/AppDelegate.swift

```diff
@@ -158,3 +158,11 @@ extension AppDelegate: MFMailComposeViewControllerDelegate {
     }
 }
 #endif
+
+extension AppDelegate: UIApplicationDelegate {
+    func applicationDidEnterBackground(application: UIApplication) {
+        let notificationCenter = NSNotificationCenter.defaultCenter()
+        var info = [NSObject: AnyObject]()
+        notificationCenter.postNotificationName("ApplicationWillTerminate", object: self, userInfo: info)
```

Comment:
I think the name is misplaced because both `IApplicationDidEnterBackgroundNotification` and `UIApplicationWillTerminateNotification` exist. This one should clearly be the former.

Actually, there is no need to send this notification manually at all. Other objects interested in the `UIApplication` lifecycle events can simply listen to the system-provided `UIApplicationDidEnterBackgroundNotification` notification which does exactly what you do here.